test/plog/LogStreamTest.cpp

```diff
@@ -0,0 +1,449 @@
+            return seastar::when_all_succeed(writeFutures.begin(), writeFutures.end());
+        })
+        // Read all the data from reloaded log stream
+        .then([this, header] (std::vector<std::tuple<Status, String, uint32_t> >&& responses){
+            for (auto& response:responses){
+                auto& [status, plogId, appended_offset] = response;
+                if (!status.is2xxOK()){
+                    throw std::runtime_error("cannot append data!");
+                }
+                K2LOG_D(log::ltest, "{}, {}", plogId, appended_offset);
+            }
+            K2LOG_I(log::ltest, ">>> Test3.3: Write Done");
+            return _reload_logStream->read(_initPlogId, 0, 6000*10005);
+        })
+        // read with continuation
+        .then([this] (auto&& response){
+            uint32_t request_size = 6000*10005;
+            auto& [status, continuation_token, payload] = response;
+            if (!status.is2xxOK()){
+                throw std::runtime_error("cannot read data!");
+            }
+            request_size -= payload.getSize();
+            Payload read_payload;
+            for (auto& b: payload.shareAll().release()) {
+                read_payload.appendBinary(std::move(b));
+            }
+            
+            return seastar::do_with(std::move(read_payload), std::move(request_size), std::move(continuation_token), [&] (auto& read_payload, auto& request_size, auto& continuation_token){
+                return seastar::do_until(
+                    [&] { return request_size == 0; },
+                    [&] {
+                        return _reload_logStream->read(continuation_token, request_size)
+                        .then([&] (auto&& response){
+                            auto& [status, token, payload] = response;
+                            if (!status.is2xxOK()){
+                                throw std::runtime_error("cannot read data!");
+                            }
+                            request_size -= payload.getSize();
+                            continuation_token = std::move(token);
+                            for (auto& b: payload.shareAll().release()) {
+                                read_payload.appendBinary(std::move(b));
+                            }
+                            return seastar::make_ready_future<>();
+                        });
+                    }
+                )
+                .then([&] (){
+                    return seastar::make_ready_future<Payload>(std::move(read_payload));
+                });
+            });
+        })
+        // check weather the read results is the same as we wrote before
+        .then([this, header] (auto&& payload){
+            K2LOG_I(log::ltest, ">>> Test3.4: Read Done");
+            String str;
+            uint32_t count = 0;
+            payload.seek(0);
+            while (payload.getDataRemaining() > 0){
+                payload.read(str);
+                K2EXPECT(log::ltest, str, header);
```

Comment:
You should validate the contents of this read string - 10000 '0' chars

Reply:
I think K2EXPECT(log::ltest, str, header) verified this?